cedar-lean/Cedar/Partial/Request.lean

```diff
@@ -81,8 +80,8 @@ open Cedar.Spec (EntityUID)
 def UidOrUnknown.subst (subsmap : Subsmap) : UidOrUnknown → Option UidOrUnknown
   | .known uid => some (.known uid)
   | .unknown unk => match subsmap.m.find? unk with
-    | some (.lit (.entityUID uid)) => some (.known uid)
-    | some (.unknown unk') => some (.unknown unk') -- substituting an unknown with another unknown, we'll allow it
+    | some (.value (.prim (.entityUID uid))) => some (.known uid)
+    | some (.residual (.unknown unk')) => some (.unknown unk') -- substituting an unknown with another unknown, we'll allow it
```

Comment:
In this particular case, the motivation is "because we allow it in the more general `subst` functions, e.g. `Partial.Value.subst`".  In this particular case, since `subst` is fallible (returns `Option`), we could disallow this by simply returning `None`.  In the more general cases, `subst` is not fallible, and the convenience of infallible `subst` is probably greater than the convenience of having `Subsmap` be `Map Unknown Value` would be, if I were to guess.

More high-level discussion on this topic was already discussed in [this thread](https://github.com/cedar-policy/cedar-spec/pull/325#discussion_r1606909672), when the code was introduced.  I don't think there's a strong high-level motivation, but more of "since we can, let's define things to be more powerful, and we can revert if it causes problems".